src/main/scala/com/high-performance-spark-examples/ml/CustomPipeline.scala

```diff
@@ -122,7 +122,10 @@ class SimpleIndexer(override val uid: String)
     import dataset.sparkSession.implicits._
     val words = dataset.select(dataset($(inputCol)).as[String]).distinct
       .collect()
-    new SimpleIndexerModel(uid, words)
+    val model = new SimpleIndexerModel(uid, words)
+    model.set(inputCol, $(inputCol))
```

Comment:
So `copy` is a bit different -- in `copyValues` the `extra` is an optional paramater (it has the default value of `ParamMap.empty`. If that doesn't work let me know and we can open a JIRA to fix the API on the Spark side for that.